[CI] WatcherRestartIT testEnsureWatcherDeletesLegacyTemplates failing

This seems to related to #69918. The failed tests include testWatcherRestart as well.

**Build scan:**
https://gradle-enterprise.elastic.co/s/djloumy34vwao/tests/:x-pack:qa:rolling-upgrade:v7.4.0%23twoThirdsUpgradedTest/org.elasticsearch.upgrades.WatcherRestartIT/testEnsureWatcherDeletesLegacyTemplates

**Reproduction line:**
`./gradlew ':x-pack:qa:rolling-upgrade:v7.4.0#twoThirdsUpgradedTest' -Dtests.class="org.elasticsearch.upgrades.WatcherRestartIT" -Dtests.method="testEnsureWatcherDeletesLegacyTemplates" -Dtests.seed=75857148BDB2692E -Dtests.bwc=true -Dtests.locale=de-LU -Dtests.timezone=Atlantic/Faeroe -Druntime.java=8`

**Applicable branches:**
7.16

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.upgrades.WatcherRestartIT&tests.test=testEnsureWatcherDeletesLegacyTemplates

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: not a string containing "\"watcher_state\":\"stopped\""
     but: was "{\"_nodes\":{\"total\":3,\"successful\":3,\"failed\":0},\"cluster_name\":\"v7.4.0\",\"manually_stopped\":false,\"stats\":[{\"node_id\":\"HABVI3E7QPCYZ0yEQnQ9mw\",\"watcher_state\":\"started\",\"watch_count\":0,\"execution_thread_pool\":{\"queue_size\":0,\"max_size\":0}},{\"node_id\":\"6rPU8dtMT2ueg9rmvOxKgg\",\"watcher_state\":\"started\",\"watch_count\":0,\"execution_thread_pool\":{\"queue_size\":0,\"max_size\":0}},{\"node_id\":\"zU88hVu-RpapPD5IRywWcA\",\"watcher_state\":\"stopped\",\"watch_count\":0,\"execution_thread_pool\":{\"queue_size\":0,\"max_size\":0}}]}"

  at __randomizedtesting.SeedInfo.seed([75857148BDB2692E:589F4C1C5F424D33]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.upgrades.WatcherRestartIT.lambda$ensureWatcherStarted$3(WatcherRestartIT.java:183)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1123)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1096)
  at org.elasticsearch.upgrades.WatcherRestartIT.ensureWatcherStarted(WatcherRestartIT.java:177)
  at org.elasticsearch.upgrades.WatcherRestartIT.testEnsureWatcherDeletesLegacyTemplates(WatcherRestartIT.java:139)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at java.lang.Thread.run(Thread.java:748)

```

There are other watcher related failures in the same build scans:
* https://gradle-enterprise.elastic.co/s/jij7ycx3u6ulo
* https://gradle-enterprise.elastic.co/s/djloumy34vwao

The error message is:
> java.lang.AssertionError: Failure in test setup: Failed to initialize at least 3 watcher nodes

Example here: https://gradle-enterprise.elastic.co/s/jij7ycx3u6ulo/tests/:x-pack:qa:rolling-upgrade:v7.1.0%23twoThirdsUpgradedTest/org.elasticsearch.upgrades.UpgradeClusterClientYamlTestSuiteIT/test%20%7Bp0=mixed_cluster%2F30_ml_jobs_crud%2FTest%20get%20old%20cluster%20job%7D?anchor=e30&focused-exception-line=0-0&top-execution=1

--- a/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/WatcherRestartIT.java
+++ b/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/WatcherRestartIT.java
@@ -30,9 +30,6 @@ public class WatcherRestartIT extends AbstractUpgradeTestCase {
     }
 
     public void testEnsureWatcherDeletesLegacyTemplates() throws Exception {
-        client().performRequest(new Request("POST", "/_watcher/_start"));
-        ensureWatcherStarted();
-
         if (CLUSTER_TYPE.equals(ClusterType.UPGRADED)) {
             // legacy index template created in previous releases should not be present anymore
             assertBusy(() -> {

